Nos/Views/NoteComposer/NoteComposer.swift

```diff
@@ -321,18 +370,79 @@ struct NoteComposer: View {
             Log.error("Error when posting: \(error.localizedDescription)")
         }
     }
+
+    /// Builds a JSONEvent object for a given text and key pair.
+    /// - Parameters:
+    ///   - attributedText: The text the user wrote.
+    ///   - keyPair: Key pair of the logged in user.
+    /// - Returns: A JSONEvent object suitable to be published to Nostr.
+    private func buildJSONEvent(attributedText: AttributedString, keyPair: KeyPair) -> JSONEvent {
+        var (content, tags) = noteParser.parse(attributedText: attributedText)
+
+        if let expirationTime {
+            tags.append(["expiration", String(Date.now.timeIntervalSince1970 + expirationTime)])
+        }
+
+        // Attach the new note to the one it is replying to, if any.
+        if let replyToNote = replyToNote, let replyToNoteID = replyToNote.identifier {
+            // TODO: Append ptags for all authors involved in the thread
+            if let replyToAuthor = replyToNote.author?.publicKey?.hex {
+                tags.append(["p", replyToAuthor])
+            }
+
+            // If `note` is a reply to another root, tag that root
+            if let rootNoteIdentifier = replyToNote.rootNote()?.identifier, rootNoteIdentifier != replyToNoteID {
+                tags.append(["e", rootNoteIdentifier, "", EventReferenceMarker.root.rawValue])
+                tags.append(["e", replyToNoteID, "", EventReferenceMarker.reply.rawValue])
+            } else {
+                tags.append(["e", replyToNoteID, "", EventReferenceMarker.root.rawValue])
+            }
+        }
+
+        return JSONEvent(pubKey: keyPair.publicKeyHex, kind: .text, tags: tags, content: content)
+    }
+
+    /// Creates a preview event object from what the user wrote in the editing controller.
+    /// - Parameter attributedText: Text being previewed.
+    private func createPreviewEvent(from attributedText: AttributedString) throws -> Event? {
```

Comment:
I might also suggest moving these out of this file, too. Another new type, along the lines of an `EventStore` or `EventRepository` depending on the terminology we want to use. Or even a `PreviewEventStore` or `PreviewEventRepository` or again, whatever term we want for a type that creates and deletes objects.

Reply:
I created a `PreviewEventRepository`. An `EventRepository` seemed misleading to me given that it was not something we were planning to use to store and retrieve all events. But `PreviewEventRepository` clearly communicates that its responsibility is to store the preview event (we don't care how). I added that to Dependencies.